src/main/java/com/cloudant/client/org/lightcouch/CouchDbClient.java

```diff
@@ -516,47 +516,30 @@ public InputStream executeToInputStream(HttpConnection connection) throws CouchD
         }
 
         try {
-            JsonObject errorJsonMessage = null;
-            if(cause != null && connection.getConnection().getErrorStream() != null) {
-                String errorString = IOUtils.toString(connection.getConnection()
-                        .getErrorStream(), "UTF-8");
-                errorJsonMessage = gson.fromJson(errorString, JsonObject.class);
-            }
-
-            //User errorStream to get error message if database does not exist or already exists
-            String response = "";
-            if(errorJsonMessage != null && errorJsonMessage.has("reason")) {
-                response = errorJsonMessage.toString();
-            } else {
-                response = connection.getConnection().getResponseMessage();
-            }
             code = connection.getConnection().getResponseCode();
+            response = connection.getConnection().getResponseMessage();
             // everything ok? return the stream
             if (code / 100 == 2) { // success [200,299]
                 return is;
             } else if (code == 404) {
                 throw new NoDocumentException(response, cause);
             } else if(code == 412) {
-                //Database already created
-                return is;
+                throw new PreconditionFailedException(response, cause);
             } else {
-                es = connection.getConnection().getErrorStream();
-
-                CouchDbException ex = getGson().fromJson(new InputStreamReader(es),
-                        CouchDbException.class);
-                if (ex == null && errorJsonMessage != null) {
-                    //Try again with error json message
-                    try {
-                        ex = getGson().fromJson(errorJsonMessage,
-                                CouchDbException.class);
-                        ex.setStatusCode(code);
-                    } catch (NullPointerException e) {
-                        //Create new Exception from scratch
+                CouchDbException ex = null;
+                try {
+                    es = connection.getConnection().getErrorStream();
+                    ex = getGson().fromJson(new InputStreamReader(es),
+                            CouchDbException.class);
+                } catch(Exception e) {
+                    ex = new CouchDbException(response, code);
+                } finally {
+                    if (ex == null) {

```

Comment:
I guess it depends if anything in the `try` can generate something other than an `Exception` i.e. a subclass of `Throwable` or `Error`.